File: 프로그래머스/0/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.java
import java.util.*;

class Solution {
    public List solution(int[] arr, boolean[] flag) {
        List<Integer> answer = new ArrayList<>();
        
        for (int i = 0; i < arr.length; i++) {
            if (flag[i]) {
                for (int j = 0; j < arr[i] * 2; j++) {
                    answer.add(arr[i]);
                }
            } else  {
                for (int k = 0; k < arr[i]; k++) {
                    answer.remove(answer.size() - 1);
                }
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/0/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.js
function solution(arr, flag) {
    let X = [];
    
    for (let i = 0; i < flag.length; i++) {
        if (flag[i]) {
            // true일 때: arr[i] 값을 arr[i] * 2 번 만큼 배열에 추가
            X.push(...Array(arr[i] * 2).fill(arr[i]));
        } else {
            // false일 때: 배열의 마지막에서 arr[i] 개 만큼의 원소를 제거
            X.splice(-arr[i]);
        }
    }
    return X;
}
